From a338e5af2f08ee6fddc24a6f40779f070e1d04f0 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Tue, 4 Oct 2016 17:26:36 -0700
Subject: [PATCH 2/5] Fix p0 initializer

Due to changes in the task_struct the following warning is occurs
when initializing the global p0.  Since this structure only exists
for it's address to be taken initialize it in a manor which isn't
sensitive to internal changes to the structure.

  module/spl/spl-generic.c:58:1: error: missing braces around
  initializer [-Werror=missing-braces]

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
Closes #576
(cherry picked from commit 341dfdb3fd44048a3149e8376b2a0f94da56f01a)

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

Conflicts:
	module/spl/spl-generic.c
---
 module/spl/spl-generic.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/module/spl/spl-generic.c b/module/spl/spl-generic.c
index 4d9846c..c01a646 100644
--- a/module/spl/spl-generic.c
+++ b/module/spl/spl-generic.c
@@ -53,7 +53,7 @@ EXPORT_SYMBOL(spl_hostid);
 module_param(spl_hostid, ulong, 0644);
 MODULE_PARM_DESC(spl_hostid, "The system hostid.");
 
-proc_t p0 = { 0 };
+proc_t p0;
 EXPORT_SYMBOL(p0);
 
 #if BITS_PER_LONG == 32
@@ -523,6 +523,8 @@ spl_init(void)
 {
 	int rc = 0;
 
+	bzero(&p0, sizeof (proc_t));
+
 	if ((rc = spl_kvmem_init()))
 		goto out1;
 
-- 
2.7.4

